### PR TITLE
Add Csv reader

### PR DESCRIPTION
**Pull request by allen-ng**
A simple csv reader interface based on libcsv, which will fall back on pure ruby when nmatrix-csv is not installed.

- nmatrix-csv
- NMatrix::IO::Csv

#407

### CODE REVIEW

**Comment by allen-ng** [comment]
If there is any problem please tell me, I'm ready to rewrite them. Thanks.

File: lib/nmatrix/nmatrix.rb [hunk 1]
```diff
@@ -100,6 +101,17 @@ def load_pcd_file(file_path)
       NMatrix::IO::PointCloudReader::MetaReader.new(file_path)
     end
 
+    # call-seq:
+    #     load_pcd_file(path) -> PointCloudReader::MetaReader
+    #
+    # * *Arguments* :
+    #   - +file_path+ -> The path to a PCL PCD file.
+    # * *Returns* :
+    #   - A PointCloudReader::MetaReader object with the matrix stored in its +matrix+ property
```
Comment (v0dro) [inline lib/nmatrix/nmatrix.rb]: change docs to csv :P

File: spec/io_spec.rb [hunk 2]
```diff
@@ -146,4 +146,27 @@
...
+
+  it "reads csv file not aligned properly with 0" do
+    n = NMatrix::IO::Csv.load 'spec/2x3_dense_not_aligned.csv'
+    m = NMatrix.new([2, 3], [1, 2, 0, 3, 4, 5], dtype: :float64)
+    expect(n).to eq(m)
+  end
+
+  it "fails to read malformed csv file" do
+    n = NMatrix::IO::Csv.load 'spec/malformed_csv.csv'
+    expect(n).to eq(nil)
```
Comment (v0dro) [inline spec/io_spec.rb]: Should this not raise an error for malformed CSV?

**Comment by v0dro** [comment]
I'd like to point out that you've overridden some previous commits relating to hilbert and linspace functions. Please rebase against the current master and then push your code again.

Also, why not let the user install libcsv herself through her own package manager? Why include the source in the nmatrix plugin? Things become harder to maintain that way.

I can see a debian package here: GSL::Ran::Discrete.alloc(GSL::Vector.alloc(0.59, 0.4, 0.01))

**Comment by allen-ng** [comment]
Thanks for your attention, I will rewrite them, this is my first Pull Request, :P.
I will remove that libcsv source code then and follow you advice.

And as a (Ubuntu) user, I also installed libcsv file myself, but it was installed at a wrong place against where ubuntu usually should do, and I have to modify ldconfig files, I'm not sure if this is suitable for user. Well, actually I think I should follow your advice, thanks.

P.S. what does that `debian package: GSL::Ran::Discrete.alloc(GSL::Vector.alloc(0.59, 0.4, 0.01))` mean

**Comment by v0dro** [comment]
Sorry for that. The link is here: https://packages.debian.org/sid/libcsv-dev

**Comment by v0dro** [comment]
Ideally your plugin should find out where the library is located and use it.

**Pull request closed without merging**